packages/expo-clipboard/ios/ClipboardPasteButton.swift

```diff
@@ -0,0 +1,169 @@
+import ExpoModulesCore
+import UniformTypeIdentifiers
+
+class ClipboardPasteButton: ExpoView {
+  let onPastePressed = EventDispatcher()
+
+  // MARK: - Properties
+  var baseBackgroundColor: UIColor?
+  var baseForegroundColor: UIColor?
+  var cornerStyle: CornerStyle = .capsule
+  var displayMode: DisplayMode = .iconAndLabel
+  var acceptedContentTypes: [AcceptedTypes] = []
+  var imageOptions = GetImageOptions()
+
+  private var childView: UIView?
+
+  func update() {
+    unmountChild()
+    if #available(iOS 16.0, *) {
+      mountView()
+    } else {
+      log.error("ClipboardPasteButton is only supported on iOS 16 and above")
+    }
+  }
+
+  @available(iOS 16.0, *)
+  private func mountView() {
+    let configuration = UIPasteControl.Configuration()
+    configuration.baseBackgroundColor = baseBackgroundColor
+    configuration.baseForegroundColor = baseForegroundColor
+    configuration.cornerStyle = cornerStyle.toCornerStyle()
+    configuration.displayMode = displayMode.toUIDisplayMode()
+
+    let control = UIPasteControl(configuration: configuration)
+    control.translatesAutoresizingMaskIntoConstraints = false
+    control.target = self
+    setContentTypes()
+
+    self.addSubview(control)
+    childView = control
+
+    NSLayoutConstraint.activate([
+      control.topAnchor.constraint(equalTo: topAnchor),
+      control.bottomAnchor.constraint(equalTo: bottomAnchor),
+      control.leadingAnchor.constraint(equalTo: leadingAnchor),
+      control.trailingAnchor.constraint(equalTo: trailingAnchor)
+    ])
+  }
+
+  override func paste(itemProviders: [NSItemProvider]) {
+    if #available(iOS 14.0, *) {
```

Comment:
Last nitpick: replacing it with `guard .. else` would reduce the indentation level 🙂